### PR TITLE
fix(.github/android): pass matrix.target and increase api to v26 

### PR DESCRIPTION
**Pull request by mxinden**
When extracting the emulator script into a separate file, the `matrix.target` substitution wasn't replaced with an environment variable.

Follow-up to https://github.com/quinn-rs/quinn/pull/1950.

Also the `ecn_*` unit tests are failing with any Android API version < v26 with:

```
2024-08-22T14:41:30.636494Z  WARN quinn_udp: sendmsg error: Os { code: 22, kind: InvalidInput, message: "Invalid argument" }, Transmit: { destination: 127.0.0.1:50196, src_ip: None, enc: Some(Ect0), len: 5, segment_size: None }
```

### CODE REVIEW

**Review by djc — APPROVED** [review]

File: .github/workflows/rust.yml [hunk 1]
```diff
@@ -102,10 +102,10 @@ jobs:
             emulator-arch: x86_64
             # Note that x86_64 image is only available for API 21+. See
             # https://github.com/ReactiveCircus/android-emulator-runner?tab=readme-ov-file#configurations.
-            api-level: 21
+            api-level: 26
```
Comment (djc) [inline .github/workflows/rust.yml:105]: I think this is conceptually unrelated and thus merits a separate commit.

File: .github/workflows/rust.yml [hunk 2]
```diff
@@ -102,10 +102,10 @@ jobs:
...
           - target: i686-linux-android
             emulator-arch: x86
-            api-level: 19
+            api-level: 26
```
Comment (mxinden) [inline .github/workflows/rust.yml:108]: The `ecn_*` unit tests are failing with any Android API version < v26 with:

```
2024-08-22T14:41:30.636494Z  WARN quinn_udp: sendmsg error: Os { code: 22, kind: InvalidInput, message: "Invalid argument" }, Transmit: { destination: 127.0.0.1:50196, src_ip: None, enc: Some(Ect0), len: 5, segment_size: None }
```

I will open a separate pull request discussing this failure.

I suggest we merge here already. Some Android tests (i.e. >=26) are better than none.

File: .github/workflows/rust.yml (hunk 1 above)
Comment (mxinden) [reply .github/workflows/rust.yml:105]: :+1: sorry for not doing that earlier. I updated the pull request @djc.

File: .github/workflows/rust.yml (hunk 2 above)
Comment (mxinden) [reply .github/workflows/rust.yml:108]: > I will open a separate pull request discussing this failure.

For the record, the separate pull request: https://github.com/quinn-rs/quinn/pull/1975

**Review by djc — APPROVED** [review]

**Pull request merged**